zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/model/Zone.java

```diff
@@ -5,12 +5,33 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import javax.validation.constraints.NotBlank;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @JsonInclude(Include.NON_NULL)
 public class Zone {
+
+    @NotBlank(message = org.cloudfoundry.identity.uaa.zone.ZoneController.MANDATORY_VALIDATION_MESSAGE)
     private String adminSecret;
+
+    @NotBlank(message = org.cloudfoundry.identity.uaa.zone.ZoneController.MANDATORY_VALIDATION_MESSAGE)
     private String subdomain;
+
+    public String getAdminSecret() {
+        return adminSecret;
+    }
+
+    public void setAdminSecret(String adminSecret) {
+        this.adminSecret = adminSecret;
+    }
+
```

Comment:
@Data annotation will generate getter, setter methods for adminSecret and subdomain fields so please revert this change.

Reply:
Done, Thanks!